core/ustring.cpp

```diff
@@ -2140,6 +2147,14 @@ int64_t String::to_int(const CharType *p_str, int p_len) {
 
 				if (c >= '0' && c <= '9') {
 
+					if (integer > INT32_MAX / 10) {
+						String number("");
+						str = p_str;
+						while (str != limit) {
+							number += *(str++);
+						}
+						ERR_FAIL_V_MSG(sign == 1 ? INT32_MAX : INT32_MIN, "Cannot represent " + number + " as integer, provided value is " + (sign == 1 ? "too big." : "too small."));
```

Comment:
This should be `INT64_MAX` and `INT64_MIN` since this function returns `int64_t`.

Reply:
Fixed by #32055.